### ISSUE TITLE
Style/WhileUntilModifier must copy logic from Style/IfUnlessModifier

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#8281]
There's a lot of logic in `IfUnlessModifier` that is missing from `WhileUntilModifier`.

1) Looses comments:

```ruby
while bar # important comment about to be nuked
  baz
end
# corrected to
baz while bar
```

2) Incorrect correction for assignment:

```ruby
foo = while bar
  baz
end
# corrected to
foo = baz while bar
```

This is logic that `IfUnlessModifier` has.

It should be moved into their shared module, and the corresponding tests should be moved in their shared context.

3) Doesn't detect inline cases:

```ruby
while foo; bar; end # => no offense
```

Again, `IfUnless` does the right thing. Same logic should be applied, tests should be moved in their shared context.

Found while checking #8273. See also #8282 and #8283

**Comment by marcandre** [comment]
See also #8499
